### PR TITLE
automatically detect bash or sh

### PR DESCRIPTION
**Pull request by termie**
# wat 

- Automatically detect bash or sh for containers by default,
defaulting to bash if it is there (#301)
- Fix a small bug when doing local deploys and using a working-dir other
 than .wercker (#301)

# y 

alpine containers use /bin/sh, many others use /bin/bash, our default has always been /bin/bash, but this made many containers fail until a command was set explicitly, this removes that need

re deploy bug, local deploy auto-path-lookup-magic was using a hardcoded .wrcker instead of looking at working-dir

### CODE REVIEW

File: docker/box.go [hunk 1]
```diff
@@ -331,7 +331,7 @@ func (b *DockerBox) RecoverInteractive(cwd string, pipeline core.Pipeline, step
 	env = append(env, pipeline.Env().Hidden.Export()...)
 	env = append(env, step.Env().Export()...)
 	env = append(env, fmt.Sprintf("cd %s", cwd))
-	cmd := []string{b.cmd}
+	cmd, _ := shlex.Split(b.cmd)
```
Comment (jwolski2) [inline docker/box.go]: Why ignore the error?

File: tests/projects/scratch-n-push/wercker.yml [hunk 2]
```diff
@@ -10,5 +10,5 @@ build:
         username: $DOCKER_USER
         password: $DOCKER_PASSWORD
         registry: https://quay.io
-        tag: $WERCKER_GIT_BRANCH-$WERCKER_GIT_COMMIT
+        tag: uniqueTagFromTest
```
Comment (jwolski2) [inline tests/projects/scratch-n-push/wercker.yml]: Was this an accidental modification?

Comment (termie) [inline tests/projects/scratch-n-push/wercker.yml]: i think there was an accidental modification that made this change a long time ago, the correct tag is the uniqueTagFromTest which is checked for in the test

File: docker/box.go (hunk 1 above)
Comment (termie) [inline docker/box.go]: ah, yeah here we shouldn't ignore, we ignore it in the other places because we are actually parsing a static string in those places

**Pull request merged**